rest/utilities_testing.go

```diff
@@ -81,7 +91,13 @@ type RestTester struct {
 	closed                  bool
 }
 
+// NewRestTester returns a rest tester backed by a single database and a single collection. This collection may be named or default collection based on global test configuration.
```

Comment:
When using NewRestTester(), how are the tests that use /db/ in paths resolving to a single named collection when running with SG_TEST_USE_DEFAULT_COLLECTION=false?  Are they still relying on the handler.go code to pick a single collection that needs to be removed?

Reply:
This is out of scope for this PR, but this works because if there's a single collection `foo.bar` the following are equivalent as per part of PRD. We discussed changing this:

- /db
- /db.bar
- /db.foo.bar

But I think `/db` should only be `/db._default._default` to avoid confusion, but we'll have to add new functions that get keyspaces and this code review was already really long.

Reply:
Yes, I think it makes sense to have this PR deliver "Run RestTester against a single named collection when SG_TEST_USE_DEFAULT_COLLECTION=false", and have the "fix RestTester tests that depend on /db/ mapping to /db.foo.bar/" be a separate ticket.

Reply:
Filed https://issues.couchbase.com/browse/CBG-2622 for this